src/plugins/data/public/data_sources/default_datasource/default_datasource.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObject } from '../../../../../core/public';
+import { IndexPatternSavedObjectAttrs } from '../../index_patterns/index_patterns';
+import { DataSource, DataSourceConfig, IndexPatternOption } from '../datasource';
+
+export class DefaultDslDataSource extends DataSource<
+  any,
+  any,
+  Promise<IndexPatternOption[] | undefined>,
+  any,
+  any
+> {
+  private readonly indexPatterns;
+
+  constructor({ name, type, metadata, indexPatterns }: DataSourceConfig) {
+    super(name, type, metadata);
+    this.indexPatterns = indexPatterns;
+  }
+
+  async getDataSet(dataSetParams?: any) {
```

Comment:
```suggestion
  async getDataSet() {
```

Reply:
`dataSetParams` exists based on https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a27d7e0fad8494491cff911190ca5b78f895be01/src/plugins/data/public/data_sources/datasource/datasource.ts#L55. although shouldn't be `any`